### ISSUE TITLE
confused by a formula in class_balanced_cross_entropy_loss.py

### ISSUE THREAD

**Issue by jxlijunhao** [kmaninis/OSVOS-PyTorch#12]
Hi, your repo is really awesome.

and I have a problem in understanding a formula in class_balanced_cross_entropy_loss.py
`loss_val = torch.mul(output, (labels - output_gt_zero)) - torch.log(
        1 + torch.exp(output - 2 * torch.mul(output, output_gt_zero)))`

compared with the original formula, is there any special purpose?

**Comment by scaelles** [comment]
Hello,
The balanced cross entropy loss formula as written in the paper it's prone to overflows during training. Therefore, the formula that we use it's just rearranging some factors to avoid this problem. You can find [here](https://www.tensorflow.org/api_docs/python/tf/nn/sigmoid_cross_entropy_with_logits) a similar derivation to the one that we are using. Let us know if you have any further questions!
